src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState_SourceGenerators.cs

```diff
@@ -26,52 +28,92 @@ internal sealed record SourceGeneratorMap(
     /// of things so that we don't cause source generators to be loaded (and fixed) within VS (which is .net framework
     /// only).
     /// </summary>
-    private static readonly ImmutableArray<(string language, AnalyzerReferencesToSourceGenerators referencesToGenerators, AnalyzerReferencesToSourceGenerators.CreateValueCallback callback)> s_languageToAnalyzerReferencesToSourceGeneratorsMap =
-    [
-        (LanguageNames.CSharp, new(), (static rs => ComputeSourceGenerators(rs, LanguageNames.CSharp))),
-        (LanguageNames.VisualBasic, new(), (static rs => ComputeSourceGenerators(rs, LanguageNames.VisualBasic))),
-    ];
-
-    private static SourceGeneratorMap ComputeSourceGenerators(IReadOnlyList<AnalyzerReference> analyzerReferences, string language)
-    {
-        using var generators = TemporaryArray<ISourceGenerator>.Empty;
-        var generatorToAnalyzerReference = ImmutableDictionary.CreateBuilder<ISourceGenerator, AnalyzerReference>();
+    private static readonly ConditionalWeakTable<ProjectState, SourceGeneratorMap> s_projectStateToSourceGeneratorsMap = new();
```

Comment:
>private [](http://example.com/codeflow?start=4&length=7)

nit: reminder from earlier PR to consider FrozenDictionary

Reply:
done.